File: zendrive-ui/src/api/models/metafile/dto/SearchRequest.ts
export class SearchRequest {
	query: string;
	page: number = 0;
	pageSize: number = 30;

	constructor(query: string, page: number, pageSize: number) {
		this.query = query;
		this.page = page;
		this.pageSize = pageSize;
	}
}

File: zendrive-api/src/main/java/com/zendrive/api/rest/models/dto/metafile/SearchRequest.java
package com.zendrive.api.rest.models.dto.metafile;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SearchRequest {
	private String query;
	private int page;
	private int pageSize;
}
